### PR TITLE
Add a way to disable metadata cache

### PR DESCRIPTION
**Pull request by brfrn169**
Please take a look!

### CODE REVIEW

File: core/src/main/java/com/scalar/db/common/TableMetadataManager.java [hunk 1]
```diff
@@ -21,7 +21,7 @@ public class TableMetadataManager {
 
   public TableMetadataManager(Admin admin, long cacheExpirationTimeSecs) {
     CacheBuilder<Object, Object> builder = CacheBuilder.newBuilder();
-    if (cacheExpirationTimeSecs > 0) {
+    if (cacheExpirationTimeSecs >= 0) {
```
Comment (brfrn169) [inline core/src/main/java/com/scalar/db/common/TableMetadataManager.java:24]: After this change, we can completely disable the metadata cache when setting `cacheExpirationTimeSecs` to `0`.

File: core/src/main/java/com/scalar/db/transaction/consensuscommit/TransactionalTableMetadataManager.java [hunk 2]
```diff
@@ -22,7 +22,7 @@ public TransactionalTableMetadataManager(
       DistributedStorageAdmin admin, long cacheExpirationTimeSecs) {
 
     CacheBuilder<Object, Object> builder = CacheBuilder.newBuilder();
-    if (cacheExpirationTimeSecs > 0) {
+    if (cacheExpirationTimeSecs >= 0) {
```
Comment (brfrn169) [inline core/src/main/java/com/scalar/db/transaction/consensuscommit/TransactionalTableMetadataManager.java:25]: Ditto.

File: core/src/main/java/com/scalar/db/common/TableMetadataManager.java (hunk 1 above)
Comment (Torch3333) [reply core/src/main/java/com/scalar/db/common/TableMetadataManager.java:24]: Without this change, when `cacheExpirationTimeSecs=0`, the `CacheLoader` would still have a default `expireAfterWrite` value ?

**Review by Torch3333 — APPROVED** [review]
LGTM, thank you!

**Review by feeblefakie — APPROVED** [review]
LGTM! Thank you!

File: core/src/main/java/com/scalar/db/common/TableMetadataManager.java (hunk 1 above)
Comment (brfrn169) [reply core/src/main/java/com/scalar/db/common/TableMetadataManager.java:24]: If `cacheExpirationTimeSecs=0` without this change, as we don't set `expireAfterWrite` value, the `CacheLoader` doesn't expire any cache entries.

So after this change,
- if we set `cacheExpirationTimeSecs` to a negative value, enabling cache but no expiration
- if we set `cacheExpirationTimeSecs` to `0`, disabling cache
- if we set `cacheExpirationTimeSecs` to a positive value, enabling cache and expiration

Comment (Torch3333) [reply core/src/main/java/com/scalar/db/common/TableMetadataManager.java:24]: Ok, thank your for the clarification. 👍

**Pull request merged**